charts/karavi-observability/templates/cert-manager.yaml

```diff
@@ -0,0 +1,135 @@
+# If the karavi-topology cert and key are provided, deploy a CA Issuer using the cert and key
+{{- if and (.Values.karaviTopology.certificateFile) (.Values.karaviTopology.privateKeyFile) }}
+{{- $certificateFileContents := .Files.Get .Values.karaviTopology.certificateFile }}
+{{- $privateKeyFileContents := .Files.Get .Values.karaviTopology.privateKeyFile }}
+apiVersion: v1
+data:
+  tls.crt: {{ $certificateFileContents | b64enc }}
+  tls.key: {{ $privateKeyFileContents | b64enc }}
+kind: Secret
+type: kubernetes.io/tls
+metadata:
+  name: karavi-topology-secret
+  namespace: {{ .Release.Namespace }}
+
+---
+
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: karavi-topology-issuer
+  namespace: {{ .Release.Namespace }}
+spec:
+  ca:
+    secretName: karavi-topology-secret
+
+---
+{{- end }}
+
+# If the otelCollector cert and key are provided, deploy a CA Issuer using the cert and key
+{{- if and (.Values.otelCollector.certificateFile) (.Values.otelCollector.privateKeyFile) }}
+{{- $certificateFileContents := .Files.Get .Values.otelCollector.certificateFile }}
+{{- $privateKeyFileContents := .Files.Get .Values.otelCollector.privateKeyFile }}
+apiVersion: v1
+kind: Secret
+type: kubernetes.io/tls
+metadata:
+  name: otel-collector-secret
+  namespace: {{ .Release.Namespace }}
+data:
+  tls.crt: {{ $certificateFileContents  | b64enc }}
+  tls.key: {{ $privateKeyFileContents | b64enc }}
+
+---
+
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: otel-collector-issuer
+  namespace: {{ .Release.Namespace }}
+spec:
+  ca:
+    secretName: otel-collector-secret
+
+---
+{{- end }}
+
+# If any set of cert+key combos are not provided, deploy a selfsigned-issuer
+{{- if or (and (not .Values.karaviTopology.certificateFile) (not .Values.karaviTopology.privateKeyFile)) (and (not .Values.otelCollector.certificateFile) (not .Values.otelCollector.privateKeyFile)) }}
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: selfsigned-issuer
+  namespace: {{ .Release.Namespace }}
+spec:
+  selfSigned: {}
+
+---
+{{- end }}
+
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: otel-collector
+  namespace: {{ .Release.Namespace }}
+spec:
+  secretName: otel-collector-tls
+  duration: 2160h # 90d
```

Comment:
What happens after 90 days? Will the certificate be re-created again by cert-manager?

Reply:
Yes, the cert will automatically be renewed.